test_helpers.py

```
@@ -137,7 +156,25 @@ def ssrf():
         return f"<pre>{resp.text}</pre>"
     except Exception as e:
         return f"<pre>Failed to fetch URL: {e}</pre>"
+def login():
```

Ruff (0.12.2)

Redefinition of unused `login` from line 112

(F811)